src/lib/Microsoft.Health.Fhir.Ingest/Telemetry/ExceptionTelemetryProcessor.cs

```diff
@@ -31,33 +30,71 @@ public ExceptionTelemetryProcessor()
         {
         }
 
-        public ExceptionTelemetryProcessor(params Type[] handledExceptionTypes)
+        public ExceptionTelemetryProcessor(params System.Type[] handledExceptionTypes)
         {
-            _handledExceptions = new HashSet<Type>(handledExceptionTypes);
+            _handledExceptions = new HashSet<System.Type>(handledExceptionTypes);
         }
 
-        public virtual bool HandleException(Exception ex, ILogger log)
+        public virtual bool HandleException(Exception ex, ITelemetryLogger log, string connectorStage)
         {
             EnsureArg.IsNotNull(ex, nameof(ex));
+            EnsureArg.IsNotNull(log, nameof(log));
+
             var exType = ex.GetType();
 
             var lookupType = exType.IsGenericType ? exType.GetGenericTypeDefinition() : exType;
 
             if (_handledExceptions.Contains(lookupType))
             {
-                if (ex is ITelemetryEvent evt)
+                if (ex is ITelemetryMetric met)
                 {
-                    log.LogMetric(name: evt.EventName, value: 1);
+                    log.LogMetric(
+                        metric: met.Metric,
+                        metricValue: 1);
                 }
                 else
                 {
-                    log.LogMetric(name: exType.Name, value: 1);
+                    var metric = ConvertExceptionToMetric(lookupType, exType);
+                    log.LogMetric(
+                        metric: metric,
+                        metricValue: 1);
                 }
 
                 return true;
             }
 
             return false;
         }
+
+        public static Metric ConvertExceptionToMetric(System.Type lookupType, Type exType)

```

Comment:
If we keep this it would be better written as a switch statement with pattern matching;
https://docs.microsoft.com/en-us/dotnet/csharp/pattern-matching

Reply:
Ended up removing this by going with the ITelemetryMetric approach